source/mqtt_client_connection.c

```diff
@@ -721,12 +721,22 @@ PyObject *aws_py_mqtt_client_connection_on_message(PyObject *self, PyObject *arg
         return NULL;
     }
 
+    Py_CLEAR(py_connection->on_any_publish);
+
+    if (callback == Py_None) {
+        Py_RETURN_NONE;
```

Comment:
`aws_mqtt_client_connection_set_on_any_publish_handler(mqtt, NULL, NULL)` ?

In case they'd previously set a callback, but now they're disabling it?

Reply:
Good catch.